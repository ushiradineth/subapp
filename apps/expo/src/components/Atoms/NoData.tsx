import React from "react";
import { Text, View } from "react-native";
import { YStack } from "tamagui";

const NoData = ({ children, background }: { children: React.ReactNode, background?: boolean }) => {
  if (background) {
    return (
      <YStack flex={1} alignItems="center" justifyContent="center" space={4}>
        <View className="flex h-full items-center justify-center">
          <Text className="text-lg font-semibold text-gray-500">{children}</Text>
        </View>
      </YStack>
    );
  }

  return (
    <View className="flex h-full items-center justify-center">
      <Text className="text-lg font-semibold text-gray-500">{children}</Text>
    </View>
  );
};

export default NoData;
