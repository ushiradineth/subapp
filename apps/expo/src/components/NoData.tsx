import React from "react";
import { Text, View } from "react-native";

const NoData = ({ children }: { children: React.ReactNode }) => {
  return (
    <View className="flex items-center h-full justify-center">
      <Text className="text-lg font-semibold text-gray-500">{children}</Text>
    </View>
  );
};

export default NoData;
