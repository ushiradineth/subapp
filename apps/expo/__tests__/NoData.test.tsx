import React from "react";
import { render, screen } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import NoData from "../src/components/Atoms/NoData";
import config from "../tamagui.config";

describe("NoData", () => {
  it("renders correct children", () => {
    const { getByTestId } = render(
      <TamaguiProvider config={config}>
        <NoData>Testing Children</NoData>
      </TamaguiProvider>,
    );

    expect(getByTestId("nodata")).toBeTruthy();

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("renders correct children with backgroup prop", () => {
    const { getByTestId } = render(
      <TamaguiProvider config={config}>
        <NoData background>Testing Children</NoData>
      </TamaguiProvider>,
    );

    expect(getByTestId("nodata-background")).toBeTruthy();

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
