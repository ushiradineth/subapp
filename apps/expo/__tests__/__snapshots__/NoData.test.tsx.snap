// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoData renders correct children 1`] = `
<View
  className="flex h-full items-center justify-center"
  testID="nodata"
>
  <Text
    className="text-lg font-semibold text-gray-500"
  >
    Testing Children
  </Text>
</View>
`;

exports[`NoData renders correct children with backgroup prop 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "justifyContent": "center",
        "minHeight": 0,
        "minWidth": 0,
        "position": "relative",
      },
    ]
  }
  testID="nodata-background"
>
  <View
    className="flex h-full items-center justify-center"
  >
    <Text
      className="text-lg font-semibold text-gray-500"
    >
      Testing Children
    </Text>
  </View>
</View>
`;
